pkg/autodiscovery/autoconfig.go

```diff
@@ -109,6 +112,38 @@ func (ac *AutoConfig) StartConfigPolling() {
 	ac.pollerActive = true
 }
 
+// StartTagFreshnessChecker checks periodically if tags associated to a service
+// are up-to-date, act as the service was restarted to reschedule potential checks
+func (ac *AutoConfig) StartTagFreshnessChecker() {
+	ac.tagFreshnessTicker = time.NewTicker(15 * time.Second) // we can miss tags for one run
+	go func() {
```

Comment:
this goroutine will never stop as the for loop has no stop condition. WDYT of adding `ac.tagFreshnessStop`, adding a case for it in the select L121, and moving the block from L176 in it?

Reply:
🙈